### DISCUSSION TITLE
DC <-> CE communication

### DISCUSSION THREAD

**Discussion by radaniba** [marklogic-community/envision#251] [Q&A]
Hi @wpaven 

I am using CE 5.4.2 with DC 5.4.1 and I noticed something.

When I create entities from DC, and I have a model with a previous version of that entity in CE, when I load the model in CE, is there a way that this model sync with the DC version of the entity to reflect these changes, or are these separate needing manual changes ?

Rad

**Comment by wpaven** [comment]
The first time you run CE it reads the hub and creates "My Hub Model" based on the existing entities in the system. So if you edit or create entities in Hub Central, you can (re)start CE with:

java -DforceInstall=true -jar hub-central-community-5.4.2.jar 

and this will re-initialize CE to recreate MyHub Model and you'll have the latest HC changes in Connect.  You can then rename the model of course to be whatever it was previously.

Hope this helps,
Pete

**Reply by radaniba** [reply]
Hi @wpaven 
Thanks for your recommendation
I tried to do the recommended approach but now I don't see the models at all in "Connect" the entities are visible, the flows are visible and working but not the graph
I am testing other functionalities

The errors I got are :

```
10:10:22.213 [http-nio-9003-exec-7] ERROR o.a.c.c.C.[.[.[.[dispatcherServlet] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is com.marklogic.client.FailedRequestException: Local message: failed to POST at /envision/model/toDatahub.sjs: Internal Server Error. Server Message: JS-JAVASCRIPT: node.properties.forEach(p => { -- Error running JavaScript request: RangeError: Maximum call stack size exceeded . See the MarkLogic server error log for further detail.] with root cause
com.marklogic.client.FailedRequestException: Local message: failed to POST at /envision/model/toDatahub.sjs: Internal Server Error. Server Message: JS-JAVASCRIPT: node.properties.forEach(p => { -- Error running JavaScript request: RangeError: Maximum call stack size exceeded . See the MarkLogic server error log for further detail.
```

**Reply by radaniba** [reply]
For the sake of update :) I upgraded Data Hub version to 5.4.3 and the behaviour is exactly like you mentioned (the nodes display stacked though but it is Ok), the error message is still there on the log, but the system works as expected

Thanks @wpaven 

Rad

**Reply by wpaven** [reply]
Glad you got it sorted.
HC and CE both expect to be "in charge" of the data hub artifacts.  So editing outside of either isn't really supported, but you can do it with workarounds like the forceInstall trick.  Eventually CE visualizations will be merged into HC and then there'll only be one app so that will make things simpler.  Though if you edit artifacts outside of the app, you can probably expect to have to take steps to keep things in synch.

**Reply by radaniba** [reply]
I love the viz part of CE, but it keeps resetting the facets on my model even with the forceInstall trick, really looking to have the graph and concept capabilities included in HC that would be extremely helpful to have one app to focus on.

thanks for your support @wpaven

**Comment by wpaven** [comment]
The plan is for the visual elements of CE not in HC to migrate into HC and then CE will eventually  be deprecated.  For now the best we have to offer is run in parallel.
